test/integration/scheduler/queue_test.go

```diff
@@ -327,6 +327,37 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			// because QHint of TaintToleration would decide to ignore a Pod update.
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name: "Pod rejected by the PreFilter of NodeAffinity plugin and Filter of PodTopologySpread is requeued when a Node is created and turned to ready",
```

Comment:
Unfortunately, this test case doesn't make sure this PR works correctly because `pod1` wouldn't get filtered out by PreFilterResult of NodeAffinity in the first place. 
Note that, NodeAffinity returns PreFilterResult only when it has `metadata.name` field selector. `NodeAffinityIn()` that crafts `pod1` doesn't do that.

I think the test should be like:

1. Prepare a Pod with NodeAffinity of "metadata.name == node1".
2. Prepare two Nodes named node1 and node2, and make node1 very small so that node1 couldn't accommodate the Pod.
    1. Then, node1 is unschedulable because of noderesourcefit plugin, and node2 is filtered out by NodeAffinity's PreFilterResult. And the Pod will go back to the queue with `UnschedulablePlugins: [NodeAffinity, NodeResourceFit]` _(if this PR correctly works!)_
3. In triggerFn, first we once remove node1. 
    1. This removal shouldn't make the Pod requeued because nodeaffinity plugin and noderesourcefit plugin both don't have Node/Removal event in `EventsToRegister`. 
4. In triggerFn, next we recreate node1 but again with very small resource so that node1 couldn't accommodate the Pod. With this event, we should see the pod requeued to the activeQ/backoffQ.
    1. For this Node/Create event, NodeAffinity QHint returns `Queue`, but noderesourcefit QHint returns `QueueSkip`. In other words, if this PR doesn't correctly work and the Pod doesn't have NodeAffinity in `UnschedulablePlugins`, this test would fail.

Reply:
> Note that, NodeAffinity returns PreFilterResult only when it has `metadata.name` field selector. `NodeAffinityIn()` that crafts `pod1` doesn't do that.

Sorry, I didn't look at the `PreFilter` code of  `NodeAffinityIn ` plugin carefully. Revise it as your suggestion